container/root/usr/local/bin/archstring

```diff
@@ -0,0 +1,54 @@
+#!/bin/bash
+
+# Script to detect processor architecture and output one of the two input parameters
+# Example: when a download path requires an arch-specific package name
+#
+# --x64 {value}: output if an x86_64 architecture is detected
+# --arm64 {value}: output if an arm64 or aarch64 architecture is detected
+function usage() {
+  echo "usage: $(basename $0) --x64 value --arm64 value"
+  exit 1
+}
+
+if [[ "$#" -eq 0 ]]; then
+  usage
+fi
+
+while [[ "$#" -gt 0 ]]
+do
+  case $1 in
+    -i|--x64)
+      declare x64="$2"
+      ;;
+    -a|--arm64)
+      declare arm64="$2"
+      ;;
+  esac
+  shift
+done
```

Comment:
Could tighten this a little further to error out on unrecognized command line args:
```
do
  case $1 in
    -i|--x64)
      declare x64="$2"
      ;;
    -a|--arm64)
      declare arm64="$2"
      ;;
    *)
      usage
      ;;
  esac
  # shift 2, since we consumed the arch specifier and the value
  shift 2
```

Reply:
well said.